sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -288,9 +288,10 @@ case class TruncateTableCommand(
   override def run(sparkSession: SparkSession): Seq[Row] = {
     val catalog = sparkSession.sessionState.catalog
     if (!catalog.tableExists(tableName)) {
-      logError(s"table '$tableName' in TRUNCATE TABLE does not exist.")
+      throw new AnalysisException(s"Table '$tableName' in TRUNCATE TABLE does not exist.")
```

Comment:
Sorry.... Just realized that I had a typo.... `For drop table with IF EXISTS keyword` should be `For drop table without IF EXISTS keyword`.  Seems without IF EXISTS keyword, Hive (I tried 1.2.1) does not throw an exception if the table does not exist.